### DISCUSSION TITLE
运行nonebot显示sqlalchemy缺少

### DISCUSSION THREAD

**Discussion by HX-Wrdzgzs** [nonebot/discussions#537] [问答]
开发者您好，根据官网的提示我安装了数据库，并且在nb orm check中发现了sqlalchemy缺少，随后安装完成后再次运行仍然显示错误，想问下这是哪边出了问题，谢谢
![Screenshot_20241127_231129_com_microsoft_rdc_androidx_SessionActivity](https://github.com/user-attachments/assets/ac4f0358-8af4-4dc4-aa27-f7ebcf6df39e)

**Comment by yanyongyu** [comment]
你的sqlalchemy database url是不是配置错了，看上去dialect写错了，请根据你的数据库类型填写对应的dialect

**Comment by HX-Wrdzgzs** [comment]
感谢您的回答，我之前已在宝塔上部署了有关于MySQL的nonebot数据库，按照官网内容我是这么写的
![Screenshot_20241128_071557_com_hihonor_notepad_NoteEditorActivity](https://github.com/user-attachments/assets/6ebc7611-f443-4789-a04e-e2a23ff768ae)
但是这种报错跟上图一样

**Reply by yanyongyu** [reply]
你应该写 `mysql+asyncmy://localhost:3306/<database_name>`

**Reply by HX-Wrdzgzs** [reply]
想问下`SQLALCHEMY_DATABASE_URL=mysql+asyncmy://localhost:3306/<nonebot>`中要填password吗（？）

**Reply by yanyongyu** [reply]
要的，你可以参考sqla文档https://docs.sqlalchemy.org/en/20/core/engines.html#database-urls

**Reply by HX-Wrdzgzs** [reply]
感谢您的回答，但目前我还是遇到了问题，在**pip install asyncmy**之后仍会显示缺少asyncmy，并且已安装了orm
![Screenshot_20241128_231221_com_microsoft_rdc_androidx_SessionActivity](https://github.com/user-attachments/assets/aa2d02bb-9ede-4d89-b16c-0a2cec235150)
![Screenshot_20241128_230425_com_microsoft_rdc_androidx_SessionActivity_edit_7311188061793](https://github.com/user-attachments/assets/57d9a4cb-a7fe-43b0-b8f2-2679aab50e82)
![Screenshot_20241128_231104_com_microsoft_rdc_androidx_SessionActivity_edit_7301316853359](https://github.com/user-attachments/assets/e75dc68b-2aec-48db-ab25-f1347d50e1ba)

**Reply by yanyongyu** [reply]
你需要在bot的虚拟环境内安装

**Reply by HX-Wrdzgzs** [reply]
感谢您的回答，想问下bot的虚拟环境是**E:/bot/maimaiDX**吗？就像P2的文件路径

**Reply by yanyongyu** [reply]
在你的截图里应该是 `E:/bot/maimaiDX/.venv` 你需要激活这个虚拟环境，执行 venv/bin 下的 activate.bat 或 pwsh

**Reply by HX-Wrdzgzs** [reply]
谢谢您的回答，我看到了activate.bat
但是双击或者使用管理员运行闪退，是成功了吗
根据您的提示，我进入了pw，但输入activate.bat显示错误？![Screenshot_20241129_162824_com_microsoft_rdc_androidx_SessionActivity.jpg](https://github.com/user-attachments/assets/d6da5101-2aea-4710-bd06-84dcc53ad92f)

![Screenshot_20241129_162513_com_microsoft_rdc_androidx_SessionActivity.jpg](https://github.com/user-attachments/assets/e752c402-832b-4c34-813e-8263b3afd7e1)

**Reply by yanyongyu** [reply]
你需要在命令行里执行，建议你自行搜索虚拟环境使用方法

**Reply by HX-Wrdzgzs** [reply]
感谢您的回答，目前数据库问题在使用虚拟环境后已解决，但运行QQ官方机器人之后仍然遇到了问题，看上去像是公域机器人的类型没填对吗？
我在频道和群都艾特了官方机器人，返回的错误结果是一样的
麻烦您了![Image_32391307957902.png](https://github.com/user-attachments/assets/f3e97ad9-8c2a-4071-8554-24afef66fd7e)

**Reply by yanyongyu** [reply]
这个报错是你没有使用权限，请确认你是否具有按键权限

**Reply by HX-Wrdzgzs** [reply]
感谢您的回答，目前已知官方机器人需要先添加指令才能动了）
